setup.py

```diff
@@ -51,7 +51,7 @@ def get_version():
     package_dir={'': 'src'},
     python_requires='>=3.6',
     install_requires=[
-        'pydicom>=2.3.0',
+        'pydicom>=2.3.0,<2.4',
```

Comment:
`pydicom` 2.4 added some very strict tests on data types, which are not satisfied by `highdicom` calls like [this one](https://github.com/ImagingDataCommons/highdicom/blob/master/src/highdicom/seg/sop.py#L1643):
```
                frame_content_item.DimensionIndexValues = (
                    [segment_number] + index_values
                )
```